### PR TITLE
Remove reliance on bevy default features

### PR DESCRIPTION
**Pull request by snendev**
Hi! New contributing to this repo, so I hope you are OK with me submitting a PR before an issue or anything like that.

I was investigating whether sickle_ui could remove its reliance on bevy's default-features and found a few areas of cleanup in the meantime. Since all the examples seem to run fine, I thought I would submit my work here for your review. Feel free to treat it as a suggestion, or let me know what you'd like me to change.

Changes include:
- remove `bevy_reflect` and `bevy_ecs_macros` imports from crates with an existing `bevy` import
- set `default-features = false` for all bevy imports
- adds bevy features `bevy_asset`, `bevy_color`, `bevy_state`, `bevy_text`, `bevy_ui`, and `bevy_winit` (<-- this one probably is only needed in dev-dependencies to run examples but I doubt any users are replacing it in practice anyway) to relevant packages

`bevy_ui` pulls in a lot of the required features, but I thought it best to repeat the requirement for those features, in order to be a little defensive.

### CODE REVIEW

File: src/widgets/layout/resize_handles.rs [hunk 1]
```diff
@@ -1,6 +1,5 @@
 use bevy::prelude::*;
 use bevy::ui::RelativeCursorPosition;
-use bevy_reflect::Reflect;
```
Comment (eidloi) [inline src/widgets/layout/resize_handles.rs:3]: I wonder where this went? The other file imports it via `bevy::reflect::Reflect` but this files doesn't

**Comment by eidloi** [comment]
Ahoy @snendev ! Thanks for going through the cleanup, I didn't find time for it.

File: src/widgets/layout/resize_handles.rs (hunk 1 above)
Comment (snendev) [reply src/widgets/layout/resize_handles.rs:3]: it is in `bevy::prelude::*` -- wildcard imports are sneaky!

**Comment by snendev** [comment]
any time! happy to help

File: crates/sickle_ui_scaffold/src/drag_interaction.rs [hunk 2]
```diff
@@ -1,7 +1,7 @@
 use bevy::prelude::*;
+use bevy::reflect::Reflect;
```
Comment (snendev) [inline crates/sickle_ui_scaffold/src/drag_interaction.rs:2]: I should probably remove it here and other places where the wildcard import exists too, though 😅

**Review by eidloi — APPROVED** [review]

**Comment by eidloi** [comment]
Cheers!

**Pull request merged**